staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/conversion.go

```diff
@@ -352,12 +354,12 @@ func Convert_Slice_string_To_Slice_int32(in *[]string, out *[]int32, s conversio
 	return nil
 }
 
-// Convert_Slice_string_To_v1_DeletionPropagation allows converting a URL query parameter propagationPolicy
-func Convert_Slice_string_To_v1_DeletionPropagation(in *[]string, out *DeletionPropagation, s conversion.Scope) error {
+// Convert_Slice_string_To_Pointer_v1_DeletionPropagation allows converting a URL query parameter propagationPolicy
+func Convert_Slice_string_To_Pointer_v1_DeletionPropagation(in *[]string, out **DeletionPropagation, s conversion.Scope) error {
```

Comment:
is there a test demonstrating what bug this change is fixing?

Reply:
I'm confused now with about TestConvertSliceStringToDeletionPropagation is ok, but TestAppsGroupBackwardCompatibility throw a nil pointer panic
```
I1113 12:05:12.864854  107721 panic.go:679] DELETE /apis/apps/v1/namespaces/default/deployments/test-deployment2?propagationPolicy=Orphan: (463.225µs) 0 [Go-http-client/1.1 127.0.0.1:44308]
E1113 12:05:12.865064  107721 runtime.go:76] Observed a panic: runtime error: invalid memory address or nil pointer dereference
goroutine 174233 [running]:
k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters.(*timeoutHandler).ServeHTTP.func1.1(0xc13aecac60)
	/home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apiserver/pkg/server/filters/timeout.go:108 +0x107
panic(0x3cae260, 0x753b9d0)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/apis/meta/v1.Convert_Slice_string_To_Pointer_v1_DeletionPropagation(...)
	/home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/apis/meta/v1/conversion.go:360
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/apis/meta/v1.autoConvert_url_Values_To_v1_DeleteOptions(0xc018adab48, 0xc10906b080, 0x50550e0, 0xc147596ff0, 0x43a55a0, 0x43a55a0)
	/home/prow/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/apis/meta/v1/zz_generated.conversion.go:323 +0x183
```
still trying to fix it.

Reply:
propagationPolicy was properly parsed from URLs before...  can you describe what led you to change this method?

Reply:
original method won't auto generate  Convert_Slice_string_To_v1_DeletionPropagation in autoConvert_url_Values_To_v1_DeleteOptions method, and DeleteOptions has a member of *DeletionPropagation rather than DeletionPropagation

Reply:
PropagationPolicy is of type `*DeletionPropagation` - that change actually seems to be a valid one.

That said the way how it works seems incorrect to me - it should look more like:
`Convert_Slice_string_To_Pointer_v1_Time`:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/conversion.go#L267

Reply:
While I prefer DeletionPropagation as PropagationPolicy

Reply:
verified both master and this PR populate the value correctly when parsed from URL